Added the support for the spider, spot, LT24, captur and skylines on the master branch

acasadoalonso opened the pull request:
All the code added to the routines parser_spider.py, etc, ...
Seems to be working OK except the test ...

Meisterschueler commented:
Hi Angel,

thank you for the PR. The tests fail, so your code (or the tests) are buggy. Could you please fix this? I just had a look at your code. Some ideas:
1. please do not return "None" if one of the "new" parsers (dstcall != aprs) fail. The only situation where a None is correct is when the input parameter is empty.
2. I think we should not alter the "old regex" string (for dstcall == aprs). You added a second "id..." part. We have the problem, that with the new beacons we have many different id styles (ogn with 6 digits with stealth and type, lt24 5 digits, naviter 10 digits, skylines 4 digits, spot ? digits...). If we take the old regex and modify it, it works somehow. But I think it is better if we create different regex for different dstcalls. This is also nessecary because we do not have a "meta format". LT24 sends "GPS", spider sends "LWE 3D", spot sends "SPOT3 GOOD", ... So it is not possible to use just one regex for all dstcalls.

Best regards
Konstantin

Meisterschueler commented on ogn/parser/parse_capture.py:
```diff
@@ -0,0 +1,2 @@
+def parse(aprs_comment):
+    return None
```
If not implemented, thow a NotImplementedError

Meisterschueler commented on ogn/parser/parse_lt24.py:
```diff
@@ -1,2 +1,12 @@
+    if ac_match:
+        return {'address': ac_match.group('deviceID'), 'climb_rate': int(ac_match.group('climb_rate')) * fpm2ms if ac_match.group('climb_rate') else None}
+    else:
+        return None
```
If the parser fails, thow an error

Meisterschueler commented on ogn/parser/parse_skylines.py:
```diff
@@ -1,2 +1,12 @@
+    if ac_match:
+        return {'address': ac_match.group('deviceID'), 'climb_rate': int(ac_match.group('climb_rate')) * fpm2ms if ac_match.group('climb_rate') else None}
+    else:
+        return None
```
If the parser fails, throw an error.

acasadoalonso commented:
Konstantin,
I updated the parser routines as you requested and I added the test scripts, however I do not understand how the test works very well, I must be missing something.
Can you check it out and fix it.
Thanks.
AC/.

Meisterschueler commented:
Hello Angel,

I am very sorry, but all your tests fail. And the tests for the beacons I took from glidernet/ogn-aprs-protocol also fail. So I cannot merge your code. Do you have any detailed specification of the new beacons? At the moment I do not have any time to implement the parsers, but with a good specification it should be simple enough for a rainy day.

Best regards
Konstantin

acasadoalonso commented:
Konstantin,

A discussion group with the specs can be find it at :

https://github.com/glidernet/ogn-aprs-protocol

feel free to contribute.

Thanks for your help.
AC/.

On Wed, Oct 25, 2017 at 8:48 AM, Meisterschueler <notifications@github.com>
wrote:

> Hello Angel,
>
> I am very sorry, but all your tests fail. And the tests for the beacons I
> took from glidernet/ogn-aprs-protocol also fail. So I cannot merge your
> code. Do you have any detailed specification of the new beacons? At the
> moment I do not have any time to implement the parsers, but with a good
> specification it should be simple enough for a rainy day.
>
> Best regards
> Konstantin
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/glidernet/python-ogn-client/pull/41#issuecomment-339231710>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AHw8Z8OtY3QypNqZw2cehURk-DA4dZ30ks5svtnXgaJpZM4P627R>
> .
>



-- 
Angel Casado

Meisterschueler commented:
Okay, if we have the specifications for new beacons fixed and documented, we could implement the parsers.

Best regards
Konstantin

acasadoalonso commented:
Gracias ...

On Thu, Oct 26, 2017 at 10:24 AM, Meisterschueler <notifications@github.com>
wrote:

> Okay, if we have the specifications for new beacons fixed and documented,
> we could implement the parsers.
>
> Best regards
> Konstantin
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/glidernet/python-ogn-client/pull/41#issuecomment-339590202>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AHw8Zx-znCxslsVh3T7knLnBqkeWwOIXks5swEGmgaJpZM4P627R>
> .
>



-- 
Angel Casado
